bee-api/bee-rest-api/src/endpoints/routes/api/v2/message_metadata.rs

```diff
@@ -1,56 +1,35 @@
-// Copyright 2020-2021 IOTA Stiftung
+// Copyright 2020-2022 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-use std::net::IpAddr;
-
+use axum::{
+    extract::{Extension, Json, Path},
+    response::IntoResponse,
+    routing::get,
+    Router,
+};
 use bee_message::{payload::Payload, semantic::ConflictReason, MessageId};
-use bee_runtime::resource::ResourceHandle;
-use bee_tangle::Tangle;
-use warp::{filters::BoxedFilter, reject, Filter, Rejection, Reply};
 
 use crate::{
-    endpoints::{
-        config::ROUTE_MESSAGE_METADATA, filters::with_tangle, path_params::message_id, permission::has_permission,
-        rejection::CustomRejection, storage::StorageBackend, CONFIRMED_THRESHOLD,
-    },
+    endpoints::{error::ApiError, storage::StorageBackend, ApiArgsFullNode, CONFIRMED_THRESHOLD},
     types::{dtos::LedgerInclusionStateDto, responses::MessageMetadataResponse},
 };
 
-fn path() -> impl Filter<Extract = (MessageId,), Error = warp::Rejection> + Clone {
-    super::path()
-        .and(warp::path("messages"))
-        .and(message_id())
-        .and(warp::path("metadata"))
-        .and(warp::path::end())
-}
-
-pub(crate) fn filter<B: StorageBackend>(
-    public_routes: Box<[String]>,
-    allowed_ips: Box<[IpAddr]>,
-    tangle: ResourceHandle<Tangle<B>>,
-) -> BoxedFilter<(impl Reply,)> {
-    self::path()
-        .and(warp::get())
-        .and(has_permission(ROUTE_MESSAGE_METADATA, public_routes, allowed_ips))
-        .and(with_tangle(tangle))
-        .and_then(message_metadata)
-        .boxed()
+pub(crate) fn filter<B: StorageBackend>() -> Router {
+    Router::new().route("/messages/:message_id/metadata", get(message_metadata::<B>))
 }
 
 pub(crate) async fn message_metadata<B: StorageBackend>(
-    message_id: MessageId,
-    tangle: ResourceHandle<Tangle<B>>,
-) -> Result<impl Reply, Rejection> {
-    if !tangle.is_confirmed_threshold(CONFIRMED_THRESHOLD) {
-        return Err(reject::custom(CustomRejection::ServiceUnavailable(
-            "the node is not synchronized".to_string(),
-        )));
+    Path(message_id): Path<MessageId>,
+    Extension(args): Extension<ApiArgsFullNode<B>>,
+) -> Result<impl IntoResponse, ApiError> {
+    if !args.tangle.is_confirmed_threshold(CONFIRMED_THRESHOLD) {
+        return Err(ApiError::ServiceUnavailable("the node is not synchronized".to_string()));
     }
 
-    match tangle.get(&message_id).await.map(|m| (*m).clone()) {
+    match args.tangle.get(&message_id).await.map(|m| (*m).clone()) {
         Some(message) => {
             // existing message <=> existing metadata, therefore unwrap() is safe
```

Comment:
```suggestion
            // Panic: existing message <=> existing metadata, therefore unwrap() is safe.
```
Sorry...so nit-picky of me :trollface: You wanna go even more mathy?
(message exists <=> metadata exists) => unwrap save. :smile:

Reply:
haha good one 😃

Reply:
I think you can get both with `get_message_and_metadata` or something.

Reply:
seems like that code is only in https://github.com/iotaledger/bee/blob/mainnet-develop-0.4/bee-tangle/src/tangle.rs#L357-L363 branch, not shimmer-develop yet

so i guess i will change the comment as Alex suggested